File: src/main/webapp/js/Integration/Condition.js
var Reference = require("./Reference");
var Coding = require("./Coding");

function Condition(resourceType, id, patientId, encounterId, practitionerId,
	code, text, clinicalStatus, onsetDateTime) {
	this.resourceType = resourceType;
	this.id = id;
	this.patientId = patientId;
	this.encounterId = encounterId;
	this.practitionerId = practitionerId;
	this.code = code;
	this.text = text;
	this.clinicalStatus = clinicalStatus;
	this.onsetDateTime = onsetDateTime;
};

Condition.prototype.prettyPrint = function() {
	return "Resource Type: " + this.resourceType
	+ " ID: " + this.id
	+ " Patient ID: "+this.patientId
	+ " Encounter ID: "+this.encounterId
	+ " Practitioner ID: "+this.practitionerId
	+ " Code: "+JSON.stringify(this.code);
	+ " Text: "+this.text
	+ " Clinical Status: "+this.clinicalStatus
	+ " Onset Date & Time: "+this.onsetDateTime;
};

Condition.prototype.jsonPrint = function() {
	var slash = "/";
	return {	resourceType: this.resourceType,
		id: this.id,
		patient: new Reference("Patient" + slash + this.patientId),
		encounter: new Reference("Encounter" + slash + this.encounterId),
		asserter: new Reference("Practitioner" + slash + this.practitionerId),
		code: { coding: [this.code],  text: this.text},
		clinicalStatus: this.clinicalStatus,
		onsetDateTime: this.onsetDateTime
	};
};

Condition.prototype.getConditionByText = function getConditionByText(text) {
	var condition = JSON.parse(text);
	var resourceType = condition.resourceType;
	var id = condition.id;
	var patientId = condition.patient.reference.replace("Patient/","");
	var encounterId = condition.encounter.reference.replace("Encounter/","");
	var practitionerId = condition.asserter.reference.replace("Practitioner/","");
	var code = new Coding(condition.code.coding[0].system, condition.code.coding[0].code, condition.code.coding[0].display);
	var clinicalStatus = condition.clinicalStatus;
	var onsetDateTime = condition.onsetDateTime;
	return new Condition(resourceType, id, patientId, encounterId, practitionerId, code, condition.code.text, clinicalStatus, onsetDateTime);
};

module.exports = Condition;

File: src/main/java/Entities/Condition.java
package Entities;

import org.json.*;

public class Condition extends Entity {

	private static final String TEXT = "text";
	private static final String PRACTITIONER = "Practitioner";
	private static final String ASSERTER = "asserter";
	private static final String ONSET_DATE_TIME = "onsetDateTime";
	private static final String CLINICAL_STATUS = "clinicalStatus";
	private static final String SLASH = "/";
	private static final String PATIENT_LOWER = "patient";
	private static final String PATIENT_UPPER = "Patient";
	private static final String ENCOUNTER_LOWER = "encounter";
	private static final String ENCOUNTER_UPPER = "Encounter";
	private static final String CODE = "code";
	private static final String CODING = "coding";

	public Condition(String resourceType, String id, String patientId,
			String encounterId, String practitionerId,
			Coding coding, String text, String clinicalStatus, String onsetDateTime) {
		super(resourceType, id, null);

		entity.put(PATIENT_LOWER, new Reference(PATIENT_UPPER + SLASH + patientId).getJSONObject());
		entity.put(ENCOUNTER_LOWER, new Reference(ENCOUNTER_UPPER + SLASH + encounterId).getJSONObject());
		entity.put(ASSERTER, new Reference(PRACTITIONER + SLASH + practitionerId).getJSONObject());

		JSONObject codeJson = new JSONObject();
		JSONArray codeArray = new JSONArray();
		codeArray.put(coding.getJSONObject());
		codeJson.put(CODING, codeArray);
		codeJson.put(TEXT, text);
		entity.put(CODE, codeJson);
		
		entity.put(CLINICAL_STATUS, clinicalStatus);
		entity.put(ONSET_DATE_TIME, onsetDateTime);
	}
	
	public Condition(String json) {
		super(null, null, json);
	}
}
